docs/alerts.md

```diff
@@ -0,0 +1,183 @@
+| |90|60| |90|60|unknown|
+| |100|30| |100|30|unknown|
+| |110|25| |110|25|unknown|
+| |120|10| |120|10|ok|
+
+In this example, the level does not get reset to ok until it passes the `ok` handler.
+This not only requires the `crit` and `warn` checks to return false, but also requires the `ok` check to affirmatively say that things are ok.
+In this way, it is possible to delay the ok signal to the notification handler.
+
+(Note): The above drops miscellaneous columns that do not affect the result.
+
+(Note): The above levels would be integers, but are represented as strings for readability.
+
+#### Caveats
+
+The above can be implemented in pure Flux and it is unclear if it would be better to specify the input format for notifications and leave it to the user to create the appropriate map call.
+
+### Notify
+
+The `notify` function will have a common template as described here, but it will differ based on the specific notification handler.
+
+The `notify` function will take an input with at least `_time` and `_level` as columns.
+It will then notify the third party endpoint with the status of the last entry in the table which would be the most recent time.
+
+The notification handler will ignore the `unknown` level (or non-positive level values).
+
+The output of the notification handler will be the action that was (or would have been) performed for each row.
+For efficiency, it isn't required that the notification handler actually do all of the actions.
+Just that it provide a record that it would have done something if a later action didn't overwrite the earlier action.
+
+Example output based on the input from example 2 from above:
+
+|_time|_at|_action|_status|_changed|
+|-----|---|-------|-------|--------|
+|20|30|notify|crit|true|
+|30|40|notify|crit|false|
+|40|50|notify|warn|true|
+|50|60|notify|crit|true|
+|60|70|notify|warn|true|
+|70|80|notify|warn|false|
+|120|130|resolve|ok|true|
+
+As can be noticed, the handler did not do anything with the `unknown` levels and so there was nothing to log.
+The `_at` time is also 10 seconds after the time the event occured.
+This is because it is the time the action was _performed_ rather than when the check happened.
+This assumes that the table is triggered upon receiving the next point in the sequence so it is triggered
+on receiving the next point after 10 seconds.
+
+If we were in a situation where the handler were to skip an action because a future notification pre-empted it (such as if only a single table were received and the handler noticed that the alert was already resolved), the `_at` time would be null since the action never occurred, but the row would still be present.
+
+### Writing to Buckets
+
+Each of the above assumes it would be possible to write the output to a bucket and to have the bucket be the source rather than chained together in the same script.
+
+I am not sure if we should build that functionality directly into the checks or how to integrate this writing.
```

Comment:
We should create helper functions for this behavior. Maybe we have a `_check` which is what you have described and `check` which wraps it with calls to `to`. Similar for `notify` and `from`.

Reply:
I agree. I've been thinking that either we should have a parameter that adds the behavior and we would have the influxdb version of flux set the behavior for the default writer or just making sure that there's a `to()` function that will ensure the output is written since I know `to()` can get confusing to use.

I'd like to know a bit more about how it would get used though. If this were to be used by tasks, then tasks would be able to generate the appropriate flux for writing the result.

Reply:
I'd like to leverage the composability of Flux and let the user chain a `to` transformation.

Reply:
If `to()` is confusing, let's simplify it. I would prefer the writing be handled by `to()` and not encapsulated inside of `check` and `notify`.